### PR TITLE
Ignore blocked users in whoReacted avatars

### PR DESCRIPTION
**Pull request by lannoene**
yeah that's it. i took the relationship store thing from noblockedmessages and there they used a try catch when getting the blocked status for users. i haven't had any issues without it though. if i should add it, feel free to tell me.

### CODE REVIEW

File: src/plugins/whoReacted/index.tsx [hunk 1]
```diff
@@ -155,7 +156,7 @@ export default definePlugin({
         }, [message.id]);
 
         const reactions = getReactionsWithQueue(message, emoji, type);
-        const users = Object.values(reactions).filter(Boolean) as User[];
+        const users = Object.values(reactions).filter((v) => {return v && !RelationshipStore.isBlocked(v.id)}) as User[];
```
Comment (Vendicated) [inline src/plugins/whoReacted/index.tsx:159]:
```suggestion
        const users = Object.values(reactions).filter((v) => v && !RelationshipStore.isBlocked(v.id)) as User[];
```

File: src/plugins/whoReacted/index.tsx [hunk 2]
```diff
@@ -100,7 +101,7 @@ function handleClickAvatar(event: React.MouseEvent<HTMLElement, MouseEvent>) {
 export default definePlugin({
     name: "WhoReacted",
     description: "Renders the avatars of users who reacted to a message",
-    authors: [Devs.Ven, Devs.KannaDev, Devs.newwares],
+    authors: [Devs.Ven, Devs.KannaDev, Devs.newwares, Devs.lannoene],
```
Comment (Vendicated) [inline src/plugins/whoReacted/index.tsx:104]: don't add yourself as author for such minor changes

File: src/plugins/whoReacted/index.tsx [hunk 3]
```diff
@@ -29,6 +29,7 @@ import { Message, ReactionEmoji, User } from "discord-types/general";
 
 const UserSummaryItem = findComponentByCodeLazy("defaultRenderUser", "showDefaultAvatarsForNullUsers");
 const AvatarStyles = findByPropsLazy("moreUsers", "emptyUser", "avatarContainer", "clickableAvatar");
+const RelationshipStore = findByPropsLazy("getRelationships", "isBlocked");
```
Comment (Vendicated) [inline src/plugins/whoReacted/index.tsx:32]: would be a good idea to move this to webpack commons since it's used in multiple places

**Review by Vendicated — COMMENTED** [review]
This behaviour should definitely be toggleable or only apply if Hide Blocked users is enabled

**Comment by sadan4** [comment]
this should really be an option, not hardcoded

**Comment by Vendicated** [comment]
i think it would be way more intuitive to just depend on whether HideBlockedUsers is enabled or not instead of adding a setting

**Comment by lannoene** [comment]
@Vendicated Could you link me to where HideBlockedUsers is located? I'm having trouble finding it. Thanks!

**Pull request closed without merging**